File: benchmark/polars/group_by.py
import datetime
import numpy as np
import polars as pl

# ------------------------------------------------------------------------------

df = pl.read_csv("./data/housing.csv")

result = (
    df
    .group_by("ocean_proximity")
    .agg([
        pl.col("median_house_value").min().alias("Minimum_median_house_value"),
        pl.col("median_house_value").max().alias("Maximum_median_house_value")
    ])
)

print(result)

File: benchmark/explorer/group_by.exs
alias Explorer.DataFrame

require DataFrame

df = DataFrame.from_csv!("./data/housing.csv")

agg_df =
  df
  |> DataFrame.group_by("ocean_proximity")
  |> DataFrame.summarise(
    min_median_house_value: min(median_house_value),
    max_median_house_value: max(median_house_value)
  )

IO.inspect(agg_df)
